src/components/Common/SearchBar.tsx

```diff
@@ -1,25 +1,211 @@
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        }
+      );
+
+      setUserList(result.data as SelectedUser[]);
+
+      return result.data;
+    } catch (error) {
+      console.error("Failed to fetch users:", error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const handleClear = () => {
+    setKeywords("");
+    setUserList([]);
+  };
+
+  const handleInputChange = (newInputValue: string) => {
+    debouncedSetter(newInputValue);
+    setFocusSearchBar(true);
+    if (newInputValue === "") {
+      setUserList([]);
+    }
+  };
+
+  const onSelect = (selectedUsername: string) => {
+    setFocusSearchBar(false);
+    setKeywords("");
+    inputRef.current?.blur();
+    const path = `/${selectedUsername}`;
+    navigate(path);
+  };
+
+  useEffect(() => {
+    fetchUsers();
+  }, [keywords]);
+
   return (
-    <TextField
-      fullWidth
-      hiddenLabel
-      InputProps={{
-        startAdornment: (
-          <InputAdornment position="start">
-            <SearchRoundedIcon />
-          </InputAdornment>
-        ),
-      }}
-      placeholder={placeholder}
-      size="small"
-    />
+    <Box sx={styles.box}>
+      <Autocomplete
+        disablePortal
+        forcePopupIcon={false}
+        fullWidth
+        filterOptions={(x) => x}
+        getOptionLabel={(option) => `${option.displayName} @${option.username}`}
+        id="search"
+        inputValue={keywords}
+        ListboxProps={{ sx: styles.listBox }}
+        loading={loading}
+        onBlur={() => setFocusSearchBar(false)}
+        onInputChange={(_, newInputValue) => handleInputChange(newInputValue)}
+        onFocus={() => setFocusSearchBar(true)}
+        open={focusSearchBar}
+        openOnFocus
+        options={userList}
+        renderInput={(params) => (
+          <TextField
+            {...params}
+            fullWidth
+            hiddenLabel
+            InputProps={{
+              ...params.InputProps,
+              endAdornment: keywords && (
+                <InputAdornment position="end">
+                  <IconButton onClick={() => handleClear()}>
+                    <ClearIcon fontSize="small" />
+                  </IconButton>
+                </InputAdornment>
+              ),
+              startAdornment: (
+                <InputAdornment position="start">
+                  <IconButton
+                    disabled
+                    sx={
+                      focusSearchBar
+                        ? styles.searchIconFocused
+                        : styles.searchIcon
+                    }
+                  >
+                    <SearchRoundedIcon />
+                  </IconButton>
+                </InputAdornment>
+              ),
+              type: "text",
+              inputRef: inputRef,
+              onKeyDown: (e) => {
+                if (e.key === "Enter") {
+                  e.stopPropagation();
+                }
+              },
+            }}
```

Comment:
Having this stop propagation makes it so that you can't use the arrow keys to select an option from the dropdown menu, is that the behavior we'd like?